nexus-sdk/src/main/java/io/nexusrpc/OperationHandler.java

```diff
@@ -0,0 +1,228 @@
+package io.nexusrpc;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.function.BiFunction;
+import org.jspecify.annotations.Nullable;
+
+/**
+ * Handler for an operation.
+ *
+ * <p>An instance of this must be returned from an {@link OperationImpl} annotated method in a
+ * {@link ServiceImpl}. Simple synchronous operations can use {@link #sync}, more complex or
+ * asynchronous operations can manually implement this interface.
+ *
+ * @param <T> The parameter type of the operation. This can be {@link Void} for no parameter.
+ * @param <R> the return type of the operation. This can be {@link Void} for no return.
+ */
+public interface OperationHandler<T, R> {
```

Comment:
nit: Can we use `I, O` for input and output? Not sure what the Java style guides recommend here.

Reply:
`T`,  and `R` are more common I'd say and align more closely with the Java SDK